### PR TITLE
Changes_for_using_hint_extention_feature

### PR DESCRIPTION
**Pull request by YairVaknin-starkware**
# TITLE
Changes_for_using_hint_extention_feature

## Description
These changes are needed, so I could make use of the hint extention feature, when trying to load a program with hints that is nested withing a program.

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 * chore: bump pip `cairo-lang` 0.13.3 [#1884](https://github.com/lambdaclass/cairo-vm/pull/1884)
 
 #### [2.0.0-rc1] - 2024-11-20
+* chore: [#1855](https://github.com/lambdaclass/cairo-vm/pull/1880):
+  * Refactor vm crate to make it possible to use hint extension feature for nested programs with hints.
```
Comment (FrancoGiachetta) [inline CHANGELOG.md:9]: These lines should be in the "Upcoming Changes" section

**Review by FrancoGiachetta — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (YairVaknin-starkware) [reply CHANGELOG.md]: Done.

**Review by JulianGCalderon — COMMENTED** [review]
Hi @YairVaknin-starkware! Why is the new trait method `as_any_mut` necessary?

**Comment by YairVaknin-starkware** [comment]
> Hi @YairVaknin-starkware! Why is the new trait method `as_any_mut` necessary?

Hi, @JulianGCalderon. I managed without it in the end. Removed.

File: vm/src/tests/run_deprecated_contract_class_simplified.rs [hunk 2]
```diff
@@ -16,6 +16,7 @@
 */
 use crate::stdlib::{collections::HashMap, prelude::*};
 
+use crate::stdlib::any::Any;
```
Comment (JulianGCalderon) [inline vm/src/tests/run_deprecated_contract_class_simplified.rs:19]: Great @YairVaknin-starkware! Could you remove this unused dependency then?

Comment (YairVaknin-starkware) [reply vm/src/tests/run_deprecated_contract_class_simplified.rs]: Done

**Review by FrancoGiachetta — APPROVED** [review]

**Review by JulianGCalderon — APPROVED** [review]

**Pull request merged**